### ISSUE TITLE
file adapter relative paths should be resolved relative to juttle program location

### ISSUE THREAD

**Issue by mstemm** [juttle/juttle-service#74]
Steve ran into a problem where he had a juttle program using the file adapter with a relative pathname.

The juttle program and file he wanted to read were in the same directory, and the directory was below the juttle-engine root. However, the relative pathname in the 'read file' wasn't being resolved relative to the location of the juttle program, and as a result the file couldn't be read.

To really address this we'd have to do some cleanup steps where relative pathnames for the file adapter have the same path resolution that currently happens for file based module imports. This is a bit harder as juttle-service can run in both file and rendezvous modes and things like "the file for the juttle program" will only exist for file modes.

Nonetheless this is something we should try to address. Maybe the solution is to always resolve relative paths for read relative to the juttle-service root.

**Comment by demmer** [comment]
I don't think we should try to have special semantics for the string that's passed into the `-file` option for read. Not only is this actually challenging to do given that there are no special semantics attached to options to `read` or any other proc, nor should there be IMO, but moreover it's not even clear what the "correct" semantics should be. For example, given module `a` imports a sub from module `b` in which one of the arguments to the sub is the path from which to read, should the path string be interpreted in the context of the invoking module `a` or the module that contains the read `b`.

If we decide that being able to read files relative to the location of the juttle program is an important use case to support, then a better solution IMO would be to introduce some kind of `__dirname` or similar that would be filled in at parse time to contain the location of the calling module, like is done in node.js and python.
